pkg/asset/manifests/aws/zones.go

```diff
@@ -0,0 +1,272 @@
+// setting the desired zones from install-config.yaml in the managed
+// VPC deployment, when specified, otherwise default zones are discovered
+// from AWS API.
+// The CIDR blocks are calculated leaving free blocks to allow future expansions,
+// in Day-2, when desired.
+func setZonesManagedVPC(in *zonesInput) error {
+	out, err := extractZonesFromInstallConfig(in)
+	if err != nil {
+		return fmt.Errorf("failed to get availability zones: %w", err)
+	}
+
+	isPublishingExternal := in.InstallConfig.Config.Publish == types.ExternalPublishingStrategy
+	mainCIDR := capiutils.CIDRFromInstallConfig(in.InstallConfig)
+	in.Cluster.Spec.NetworkSpec.VPC = capa.VPCSpec{
+		CidrBlock: mainCIDR.String(),
+	}
+
+	// Base subnets considering only private zones, leaving one free block to allow
+	// future subnet expansions in Day-2.
+	numSubnets := len(out.AvailabilityZones) + 1
+
+	// Public subnets consumes one range from base blocks.
+	if isPublishingExternal {
+		numSubnets++
+	}
+
+	privateCIDRs, err := utilscidr.SplitIntoSubnetsIPv4(mainCIDR.String(), numSubnets)
+	if err != nil {
+		return fmt.Errorf("unable to retrieve CIDR blocks for all private subnets: %w", err)
+	}
+
+	var publicCIDRs []*net.IPNet
+	if isPublishingExternal {
+		publicCIDRs, err = utilscidr.SplitIntoSubnetsIPv4(privateCIDRs[len(out.AvailabilityZones)].String(), len(out.AvailabilityZones))
+		if err != nil {
+			return fmt.Errorf("unable to retrieve CIDR blocks for all public subnets: %w", err)
+		}
+	}
+
+	// Create subnets from zone pool with type availability-zone
+	if len(privateCIDRs) < len(out.AvailabilityZones) {
+		return fmt.Errorf("unable to define CIDR blocks to all zones for private subnets")
+	}
+	if isPublishingExternal && len(publicCIDRs) < len(out.AvailabilityZones) {
+		return fmt.Errorf("unable to define CIDR blocksto all zones for public subnets")
+	}
+
+	idxCIDR := 0
+	for zone := range out.AvailabilityZones {
+		cidr := privateCIDRs[idxCIDR]
+		in.Cluster.Spec.NetworkSpec.Subnets = append(in.Cluster.Spec.NetworkSpec.Subnets, capa.SubnetSpec{
+			AvailabilityZone: zone,
+			CidrBlock:        cidr.String(),
+			ID:               fmt.Sprintf("%s-subnet-private-%s", in.ClusterID.InfraID, zone),
+			IsPublic:         false,
+		})
+		if isPublishingExternal {
+			cidr = publicCIDRs[idxCIDR]
+			in.Cluster.Spec.NetworkSpec.Subnets = append(in.Cluster.Spec.NetworkSpec.Subnets, capa.SubnetSpec{
+				AvailabilityZone: zone,
+				CidrBlock:        cidr.String(),
+				ID:               fmt.Sprintf("%s-subnet-public-%s", in.ClusterID.InfraID, zone),
+				IsPublic:         true,
+			})
+		}
+		idxCIDR++
+	}
+	return nil
+}
+
+// extractZonesFromInstallConfig extracts zones defined in the install-config.
+func extractZonesFromInstallConfig(in *zonesInput) (*zonesCAPI, error) {
+	if in.InstallConfig == nil {
+		return nil, fmt.Errorf("unable to retrieve InstallConfig")
+	}
+	if in.InstallConfig.Config == nil {
+		return nil, fmt.Errorf("unable to retrieve Config")
+	}
```

Comment:
Could be checked once in `setZones`?

Reply:
Right. I think we can assume the installConfig is not `nil`. I can't think of a circunstance where that's not true and we're still generating a manifest :) So feel free to drop the test.